opentracing/src/main/java/opentracing/Span.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+/**
+ * Represents an in-flight span in the opentracing system.
+ *
+ * <p>Spans are created by the {@link Tracer#buildSpan} interface.
+ */
+public interface Span {
+
+  /**
+   * Sets the end timestamp and records the span.
```

Comment:
also, I think it may have been you that mentioned that the risk of
time moving backwards is an issue when doing timestamp based intervals
vs absolute + duration offset (back in the day)

I don't want this conversation to hold up mick's work as it is a
different topic which spun out of me enumerating how this is done
today in Brave. If you feel this is important enough to continue,
probably a top-level issue is best.